src/os/unix/wayland.rs

```diff
@@ -0,0 +1,1117 @@
+                        _ => {}
+                    }
+                }
+                //Event::Frame {} => {}
+                //Event::AxisSource { axis_source } => {}
+                Event::AxisStop { axis, .. } => {
+                    use wayland_client::protocol::wl_pointer::Axis;
+
+                    match axis {
+                        Axis::VerticalScroll => self.scroll_y = 0.,
+                        Axis::HorizontalScroll => self.scroll_x = 0.,
+                        _ => {}
+                    }
+                }
+                //Event::AxisDiscrete { axis, discrete } => {}
+                _ => {}
+            }
+        }
+
+        self.key_handler.update();
+    }
+
+    fn handle_key(
+        keymap: &xkb::keymap::Keymap,
+        key: u32,
+        state: wayland_client::protocol::wl_keyboard::KeyState,
+        key_handler: &mut KeyHandler,
+    ) {
+        use wayland_client::protocol::wl_keyboard::KeyState;
+
+        let is_down = state == KeyState::Pressed;
+
+        let state = keymap.state();
+        let key_xkb = state.key(key);
+
+        if let Some(keysym) = key_xkb.sym() {
+            use xkb::key;
+
+            let key_i = match keysym {
+                key::_0 => Key::Key0,
+                key::_1 => Key::Key1,
+                key::_2 => Key::Key2,
+                key::_3 => Key::Key3,
+                key::_4 => Key::Key4,
+                key::_5 => Key::Key5,
+                key::_6 => Key::Key6,
+                key::_7 => Key::Key7,
+                key::_8 => Key::Key8,
+                key::_9 => Key::Key9,
+
+                key::a => Key::A,
+                key::b => Key::B,
+                key::c => Key::C,
+                key::d => Key::D,
+                key::e => Key::E,
+                key::f => Key::F,
+                key::g => Key::G,
+                key::h => Key::H,
+                key::i => Key::I,
+                key::j => Key::J,
+                key::k => Key::K,
+                key::l => Key::L,
+                key::m => Key::M,
+                key::n => Key::N,
+                key::o => Key::O,
+                key::p => Key::P,
+                key::q => Key::Q,
+                key::r => Key::R,
+                key::s => Key::S,
+                key::t => Key::T,
+                key::u => Key::U,
+                key::v => Key::V,
+                key::w => Key::W,
+                key::x => Key::X,
+                key::y => Key::Y,
+                key::z => Key::Z,
+
+                key::apostrophe => Key::Apostrophe,
+                key::grave => Key::Backquote,
+                key::backslash => Key::Backslash,
+                key::comma => Key::Comma,
+                key::equal => Key::Equal,
+                key::bracketleft => Key::LeftBracket,
+                key::minus => Key::Minus,
+                key::period => Key::Period,
+                key::braceright => Key::RightBracket,
```

Comment:
`braceright` and `bracketright` are two different things in `xkb`. Also, it might be nice to have these "sequential" keys be next to each other (e.g. `bracketleft` and `bracketright` right above/below the other).

```suggestion
                key::bracketright => Key::RightBracket,
```